### PR TITLE
Ft 151 composition tokens pt ii

### PR DESCRIPTION
**Pull request by swordEdge**

### CODE REVIEW

File: src/app/components/SingleCompositionTokenForm.tsx [hunk 1]
```diff
@@ -1,99 +1,108 @@
...
-          <PropertySwitchMenuMainTrigger>
-            <span>{token.property}</span>
-          </PropertySwitchMenuMainTrigger>
-          <PropertySwitchMenuContent sideOffset={2}>
-            <PropertySwitchMenuRadioGroup value={token.property}>
+        <DropdownMenu open={menuOpened} onOpenChange={handleToggleMenu}>
+          <DropdownMenuTrigger css={{ flex: 3, minHeight: '$10', border: `1px solid ${error && property.length === 0 ? '$fgDanger' : '$borderMuted'}` }}>
+            <span>{property}</span>
+          </DropdownMenuTrigger>
+          <DropdownMenuContent sideOffset={2} className="content scroll-container" css={{ maxHeight: '$11', backgroundColor: '$bgDefault', color: '$text' }}>
```
Comment (six7) [inline src/app/components/SingleCompositionTokenForm.tsx:94]:
```suggestion
          <DropdownMenuContent sideOffset={2} className="content scroll-container" css={{ maxHeight: '$dropdownMaxHeight' }}>
```

No need to specify bgColor, the dropdown handles this for us. Also I changed name of token to be descriptive.

File: src/stitches.config.ts [hunk 2]
```diff
@@ -107,6 +107,7 @@ export const {
       8: '48px',
       9: '64px',
       10: '38px',
+      11: '140px',
```
Comment (six7) [inline src/stitches.config.ts:110]:
```suggestion
```

This token is only required in `sizing`

File: src/stitches.config.ts [hunk 3]
```diff
@@ -118,6 +119,7 @@ export const {
       7: '32px',
       8: '48px',
       10: '38px',
+      11: '140px',
```
Comment (six7) [inline src/stitches.config.ts:122]:
```suggestion
      dropdownMaxHeight: '170px',
```

File: src/app/components/SingleCompositionTokenForm.tsx [hunk 4]
```diff
@@ -1,99 +1,108 @@
...
       >
-        <PropertySwitchMenu open={menuOpened} onOpenChange={handleToggleMenu}>
-          <PropertySwitchMenuMainTrigger>
-            <span>{token.property}</span>
-          </PropertySwitchMenuMainTrigger>
-          <PropertySwitchMenuContent sideOffset={2}>
-            <PropertySwitchMenuRadioGroup value={token.property}>
+        <DropdownMenu open={menuOpened} onOpenChange={handleToggleMenu}>
+          <DropdownMenuTrigger css={{ flex: 3, minHeight: '$10', border: `1px solid ${error && property.length === 0 ? '$fgDanger' : '$borderMuted'}` }}>
+            <span>{property}</span>
```
Comment (six7) [inline src/app/components/SingleCompositionTokenForm.tsx:92]:
```suggestion
          <DropdownMenuTrigger bordered css={{ flex: 3, minHeight: '$10' }}>
            <span>{property || "Choose a property"}</span>
```

**Review by six7 — CHANGES_REQUESTED** [review]
Added some comments. Overall great!

**Review by six7 — APPROVED** [review]

File: src/types/CompositionTokenProperty.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { Properties } from "@/constants/Properties";
+import type { TokenTypes } from '@/constants/TokenTypes';
+import { TokenBoxshadowValue } from "./values";
+
+export type CompositionTokenProperty = keyof typeof Properties;
+
+export type CompositionTokenValue = Partial<
+Record<TokenTypes, string | number | object | Array<TokenBoxshadowValue>>
```
Comment (LiamMartens) [inline src/types/CompositionTokenProperty.ts:8]: Do not use `object`, please type it fully. `object` is too vague.

File: src/types/CompositionTokenProperty.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { Properties } from "@/constants/Properties";
```
Comment (LiamMartens) [inline src/types/CompositionTokenProperty.ts:1]: I see this is using double quotes, make sure you let eslint fix any errors before committing. Secondly, also make sure `eslint` is installed and enabled in VSCode.

File: src/utils/alias/checkIfContainsAlias.tsx [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import { checkAliasStartRegex } from '@/constants/AliasRegex';
 
-export function checkIfContainsAlias(token?: string | number | null) {
+export function checkIfContainsAlias(token?: string | number | object | null) {
```
Comment (LiamMartens) [inline src/utils/alias/checkIfContainsAlias.tsx:3]: Same comment about using `object` as a type.

File: src/app/components/TokenButton/TokenButton.tsx [hunk 8]
```diff
@@ -121,16 +119,14 @@ export const TokenButton: React.FC<Props> = ({
...
+      if (compositionToken) {
+        Object.keys(compositionToken.value).forEach((property: string) => {
+          tokensInCompositionToken[property as CompositionTokenProperty] = 'delete';
         });
-      } else {
-        tokensInCompositionToken[(compositionToken?.rawValue as TokenCompositionValue).property as keyof typeof Properties] = 'delete';
       }
       tokensInCompositionToken.composition = 'delete';
-      setPluginValue(tokensInCompositionToken);
+      setPluginValue(tokensInCompositionToken as SelectionValue);
```
Comment (LiamMartens) [inline src/app/components/TokenButton/TokenButton.tsx:129]: Are we able to fix the typing error without type casting it?

File: src/app/components/SingleCompositionTokenForm.tsx [hunk 9]
```diff
@@ -1,103 +1,107 @@
...
-  properties: string[],
-  setValue: (property: TokenCompositionValue | TokenCompositionValue[]) => void;
-  onRemove: (index: number) => void;
+  property: string;
+  value: string;
+  tokenValue: NodeTokenRefMap;
+  properties: string[];
+  setTokenValue: (neweTokenValue: NodeTokenRefMap) => void;
+  onRemove: (property: string) => void;
+  setOrderObj: (newOrderObj: object) => void;
```
Comment (LiamMartens) [inline src/app/components/SingleCompositionTokenForm.tsx:36]: Same comment, please don't use `object` since it's too vague.

**Review by LiamMartens — CHANGES_REQUESTED** [review]

**Comment by six7** [comment]
I went over changes and I'll merge seeing as @LiamMartens comments have been addressed 👍

**Pull request merged**